File: solutions/zig/pangram/1/pangram.zig
const std = @import("std");

pub fn isPangram(str: []const u8) bool {
    var letters: [26]bool = .{false} ** 26;
    for (str) |value| {
        if (std.ascii.isAlphabetic(value)) {
            letters[std.ascii.toLower(value) - 'a'] = true;
        }
    }

    for (letters) |v| {
        if (!v) {
            return false;
        }
    }

    return true;
}

File: solutions/go/pangram/1/pangram.go
package pangram

import "strings"

func IsPangram(input string) bool {
	letters := make(map[rune]bool)
	lower := strings.ToLower(input)
	for _, c := range lower {
		if c >= 'a' && c <= 'z' {
			letters[c] = true
		}
	}

	return len(letters) == 26
}
